dts/arm/atmel/saml21.dtsi

```diff
@@ -7,29 +7,205 @@
 #include <atmel/saml2x.dtsi>
 
 / {
+
+	aliases {
+		watchdog0 = &wdog;
+	};
+
+	chosen {
+		zephyr,flash-controller = &nvmctrl;
+		zephyr,entropy = &trng;
+	};
+
 	soc {
-		usb0: usb@41000000 {
-			compatible = "atmel,sam0-usb";
-			status = "disabled";
-			reg = <0x41000000 0x1000>;
-			interrupts = <6 0>;
-			num-bidir-endpoints = <8>;
+
+		nvmctrl: nvmctrl@41004000  {
+			compatible = "atmel,sam0-nvmctrl";
+			reg = <0x41004000 0x22>;
+			interrupts = <4 0>;
+			lock-regions = <16>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				reg = <0 0x40000>;
```

Comment:
flash@0 reg property should be defined on a file named saml21x18.dtsi
idem with sram0: memory@20000000, the reg property should be at saml21x18.dtsi file too.

Same principle of https://github.com/zephyrproject-rtos/zephyr/blob/32f4b6e09a85c4886767df93d8a6c4297b79b009/dts/arm/atmel/samd5xx20.dtsi#L7-L22